### PR TITLE
Add phpstan

### PR DESCRIPTION
**Pull request by alexmanno**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,10 @@ matrix:
   - php: 7.1
     env:
     - DEPS=lowest
+  - php: 7.1
+    env:
+    - EXECUTE_TEST_PHPSTAN=true
```
Comment (thomasvargiu) [inline .travis.yml:23]: We should execute it on locked dependencies

**Review by thomasvargiu — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (alexmanno) [reply .travis.yml:23]: composer.lock is not committed...

Comment (thomasvargiu) [reply .travis.yml:23]: Sorry, I meant we should not use a duplicated environment.

**Review by thomasvargiu — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -43,6 +44,7 @@ install:
 - travis_retry composer install $COMPOSER_ARGS
 
 script:
+- if [[ $EXECUTE_TEST_PHPSTAN == 'true' ]]; then ./vendor/bin/phpstan analyse -l 7 src ; fi
```
Comment (Jean85) [inline .travis.yml]: Should we run in on the test folder too? We may need phpstan-phpunit for that.

**Review by Jean85 — COMMENTED** [review]
Wow, no fixes needed, awesome!

File: .travis.yml (hunk 2 above)
Comment (thomasvargiu) [reply .travis.yml]: I agree. And it's better to call phpstan on current directory. Filters can be set on phpstan configuration if needed.

**Review by thomasvargiu — CHANGES_REQUESTED** [review]

**Review by thomasvargiu — APPROVED** [review]
Seems good IMHO

**Pull request merged**